### PR TITLE
[MM-60685] Fix racy TestSyncLdap

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
https://github.com/mattermost/mattermost/pull/28195 introduced raciness into `TestSyncLdap`. The request completes quickly, leading to a race between clearing the session Props in `PlatformService.ReturnSessionToPool` and `StartSynchronizeJob`.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-60685

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: server/channels/app/ldap.go [hunk 1]
```diff
@@ -17,20 +17,21 @@ import (
 // SyncLdap starts an LDAP sync job.
 // If includeRemovedMembers is true, then members who left or were removed from a team/channel will
 // be re-added; otherwise, they will not be re-added.
-func (a *App) SyncLdap(c request.CTX, includeRemovedMembers bool) {
+func (a *App) SyncLdap(rctx request.CTX, includeRemovedMembers bool) {
+	rctx = rctx.Clone()
```
Comment (streamer45) [inline server/channels/app/ldap.go:21]: Is this affecting production code or just to fix a test?

Comment (hanzei) [reply server/channels/app/ldap.go:21]: It's just to fix a test. `StartSynchronizeJob` does not read the session and hence doesn't race. It the comparison of the mock arguments that race.

Comment (streamer45) [reply server/channels/app/ldap.go:21]: Do you think this race could affect more (non-test) instances of `request.CTX` where we may use the session?

Comment (hanzei) [reply server/channels/app/ldap.go:21]: Potentially. Currently, this is the only test that races, which gives us an indication that the issue is not wide-spread.

Comment (streamer45) [reply server/channels/app/ldap.go:21]: If that's the case, what do you think about making the `request.Context.Session()` method return a copy of the session instead?

Comment (hanzei) [reply server/channels/app/ldap.go:21]: I'm not sure that that is strictly required. https://github.com/mattermost/mattermost/pull/28195 already eliminated a whole category of race conditions. The only ones (that we are aware of) come from request handlers that return quickly and spawn a go routine that also makes use of `rctx.Session()`. That seems to be a rare case as the only place where it happend in production is the `api/v4/websocket` endpoint.

Comment (streamer45) [reply server/channels/app/ldap.go:21]: I think this is about being proactive vs. reactive. If we know accessing from context is race-prone, I think we should guard against it. Unless there are strong concerns about performance but I am not sure that's the case here.

@agnivade Wondering if you have any thoughts on the above.

Comment (agnivade) [reply server/channels/app/ldap.go:21]: I am okay with being reactive and see if we get any further cases of this. If we do, then we can proactively clone the session every time.

@hanzei - We could change the PR temporarily to enable race mode to verify that there are no further race issues.

```
  if [[ ${{ github.ref_name }} == 'master' ]]; then
            export RACE_MODE="-race"
          fi
```

just remove the if condition.

**Review by streamer45 — APPROVED** [review]
Thanks! Good discussion :100:

File: server/channels/app/ldap.go (hunk 1 above)
Comment (hanzei) [reply server/channels/app/ldap.go:21]: I've enabled the race test one and can confirm that they run without an error.

**Pull request merged**